alf/algorithms/entropy_target_algorithm.py

```diff
@@ -35,42 +36,64 @@ class EntropyTargetAlgorithm(Algorithm):
     It tries to adjust the entropy regularization (i.e. alpha) so that the
     the entropy is not smaller than `target_entropy`.
 
-    It's described in section 5 of the following paper:
-    Haarnoja et al "Soft Actor-Critic Algorithms and Applications" arXiv:1812.05905v2
+    The algorithm has two stages:
+    1. init stage. During this stage, the alpha is not changed. It transitions
+       to adjust_stage once entropy drops below `target_entropy`.
+    2. adjust stage. During this stage, log_alpha is adjusted using this formula:
+       ((below + 0.5 * above) * decreasing - (above + 0.5 * below) * increasing) * update_rate
+       Note that log_alpha will always be decreased if entropy is increasing
+       even when the entropy is below the target entropy. This is to prevent
+       overshooting log_alpha to a too big value. Same reason for always
+       increasing log_alpha even when the entropy is above the target entropy.
+       `update_rate` is initialized to `fast_update_rate` and is reduced by a
+       factor of 0.9 whenever the entropy crosses `target_entropy`. `udpate_rate`
+       is reset to `fast_update_rate` if entropy drops too much below
+       `target_entropy` (i.e., fast_stage_thresh in the code, which is the half
+       of `target_entropy` if it is positive, and twice of `target_entropy` if
+       it is negative.
     """
 
     def __init__(self,
                  action_spec,
-                 initial_log_alpha=-3.0,
+                 initial_alpha=0.01,
                  target_entropy=None,
-                 optimizer=None,
+                 slow_update_rate=0.01,
+                 fast_update_rate=np.log(2),
                  min_alpha=1e-4,
                  debug_summaries=False):
         """Create an EntropyTargetAlgorithm
 
         Args:
             action_spec (nested BoundedTensorSpec): representing the actions.
-            initial_log_alpha (float): initial value for log(alpha). The default
-                value corresponds to alpha ~ 0.05.
+            initial_log_alpha (float): initial value for alpha.
             target_entropy (float): the lower bound of the entropy. If not
                 provided, a default value proportional to the action dimension
                 is used.
+            slow_update_rate (float):
```

Comment:
Comment on slow_update_rate (the min possible update rate)

Reply:
This means alpha is either doubled or halved.

Reply:
done